pkg/kubelet/prober/prober.go

```diff
@@ -183,6 +197,13 @@ func (pb *prober) runProbe(ctx context.Context, probeType probeType, p *v1.Probe
 	}
 }
 
+func cachedRequestStaled(p *v1.Probe, req *http.Request, status v1.PodStatus) bool {
+	if p.HTTPGet.Host == "" && req.URL.Hostname() != status.PodIP {
+		return true
+	}
+	return false
```

Comment:
```suggestion
	return p.HTTPGet.Host == "" && req.URL.Hostname() != status.PodIP {
```

Considering the suggestion, do we need a separate function for this?

Reply:
A separate function would help with readability and make the original if statement not too verbose. without it, it will look like this.
```go
		reqAny, found := pb.httpProbeReqCache.Load(key)
		if !found || (p.HTTPGet.Host == "" && reqAny.(*http.Request).URL.Hostname() != status.PodIP) {
			req, err = httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
			if err != nil {
```